Modules/_xxsubinterpretersmodule.c

```diff
@@ -347,12 +347,165 @@ channel_exceptions_init(PyObject *ns)
     return 0;
 }
 
+typedef struct _channelitem_wait_lock {
+    PyThread_type_lock lock;
+    PyThread_type_lock mutex;
+    int lock_is_allocated;
```

Comment:
The purpose of this flag is a bit unclear to me.  Why de-allocate anything before the `_channelitem_wait_lock *` is freed?  Using allocation as a flag for intermediate functionality can be confusing and sometimes a source of bugs.  I recommend changing the name to match the state for which it is actually a flag.